src/Log/Engine/BaseLog.php

```diff
@@ -61,6 +68,18 @@ public function __construct(array $config = [])
         if (!empty($this->_config['types']) && empty($this->_config['levels'])) {
             $this->_config['levels'] = (array)$this->_config['types'];
         }
+
+        $formatterConfig = $this->_config['formatter'] ?? null;
+        if (isset($formatterConfig)) {
+            $this->formatter = new $formatterConfig['className']($formatterConfig);
```

Comment:
We generally allow setting just class name strings too if no extra options are needed, so `'formatter' => Foo::class`.

Reply:
Good catch. Remembered too late.

Reply:
Could do that but that would make this one of the only class which has separate keys for class name and it's config.